src/transformers/models/blip/modeling_blip.py

```diff
@@ -0,0 +1,1453 @@
+        **generate_kwargs
+    ) -> torch.LongTensor:
+        r"""
+        Overrides *generate* function to be able to use the model as a conditional generator
+
+        Parameters:
+            input_ids (*torch.LongTensor* of shape *(batch_size, sequence_length)*):
+                The sequence used as a prompt for the generation.
+            pixel_values (*torch.FloatTensor* of shape *(batch_size, image_width, image_height)*:
+                Input image to be processed
+            attention_mask (*torch.LongTensor* of shape *(batch_size, sequence_length)*, *optional*):
+                Mask to avoid performing attention on padding token indices. Mask values selected in `[0, 1]`. `1` for
+                tokens that are NOT MASKED, `0` for MASKED tokens.
+            pixel_mask (*torch.LongTensor* of shape *(batch_size, image_width, image_height)*, *optional*):
+                Mask to be used for the image - not used but kept for compatibility with the *BlipProcessor*
+            token_type_ids (*torch.LongTensor* of shape *(batch_size, sequence_length)*, *optional*):
+                Segment token indices to indicate first and second portions of the inputs. Not used but kept for
+                compatibility with the *BlipProcessor*
+            **generate_kwargs:
+                Additional arguments passed to the *generate* function of the decoder
+
+
+        Examples:
+        ```python
+        >>> from PIL import Image
+        >>> import requests
+        >>> from transformers import BlipTokenizer, BlipForImageCaptioning
+
+        >>> model = BlipForImageCaptioning.from_pretrained("Salesforce/blip-image-captioning-base")
+        >>> processor = CLIPProcessor.from_pretrained("Salesforce/blip-image-captioning-base")
+
+        >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        >>> image = Image.open(requests.get(url, stream=True).raw)
+
+        >>> inputs = processor(images=image, return_tensors="pt")
+
+        >>> outputs = model(**inputs)
+        >>> image_embeds = outputs.image_embeds
+        ```
+        """
+        vision_outputs = self.vision_model(
+            pixel_values=pixel_values,
+        )
+
+        image_embeds = vision_outputs[0]
+
+        image_atts = torch.ones(image_embeds.size()[:-1], dtype=torch.long).to(image_embeds.device)
+        model_kwargs = {"encoder_hidden_states": image_embeds, "encoder_attention_mask": image_atts}
+
+        if isinstance(input_ids, list):
+            input_ids = torch.LongTensor(input_ids)
+
+        question_outputs = self.text_encoder(
+            input_ids=input_ids,
+            attention_mask=attention_mask,
+            encoder_hidden_states=image_embeds,
+            encoder_attention_mask=image_atts,
+            return_dict=False,
+        )
+
+        question_embeds = question_outputs[0]
+
+        question_atts = torch.ones(question_embeds.size()[:-1], dtype=torch.long).to(question_embeds.device)
```

Comment:
Better name here please

Reply:
Indeed! Fixed in [fc4b239](https://github.com/huggingface/transformers/pull/20716/commits/fc4b239c521438522b4166baa9d5decb0e121957)